### ISSUE TITLE
New Session tab submenu goes off screen 

### ISSUE THREAD

**Issue by srirambv** [brave/browser-laptop#7748]
- Did you search for similar issues before submitting this one?
Yes

- Describe the issue you encountered:
New Session tab submenu goes off screen 

- Platform (Win7, 8, 10? macOS? Linux distro?):
Windows 10 x64

- Brave Version (revision SHA):
Brave	0.14.0
rev	77c562c

- Steps to reproduce:
    1. Open bunch of tabs so  that the new tab button is close to the right end of the browser
    2. Right click/long press on new tab button to bring the context menu
    3. Navigate to New Session Tab to show the sub menu, is not contained withing the screen or over the border
    
- Actual result:
New session tab submenu is not shown completely 

- Expected result:
Should show the submenu irrespective of the position of the new tab context menu

- Will the steps above reproduce in a fresh profile? If not what other info can be added?
Yes

- Is this an issue in the currently released version?
N/A

- Can this issue be consistently reproduced?
Yes

- Extra QA steps:
    1.
    2.
    3.

- Screenshot if needed:
![image](https://cloud.githubusercontent.com/assets/17010094/23993819/fbdb8b48-0a39-11e7-81ce-2093e2503405.png)

- Any related issues:
#5763
cc: @cezaraugusto

**Comment by cezaraugusto** [comment]
I'm going to self-assign but this issue is regarding how we deal with context menus in general.

TL;DR on context menus we move the entire container together making it impossible to move nested item to the left since it's blocked by parent container offset.

Since rework is contextMenu-specific and may affect other areas such as bookmarks I'll split it to 0.14.1. Feel free to change it and undo assignment. thanks

**Comment by bsclifton** [comment]
Moving to 0.14.3. Redoing this using native context menus would be a great way to solve the issue. Currently, they block the UI on macOS (which will freeze any video playing) but otherwise work great

**Comment by alexwykoff** [comment]
@Liunkae we thought you might like to try this with a native menu. 
Want to give it a stab?

**Comment by Liunkae** [comment]
PR: #8397

**Comment by bsclifton** [comment]
Pulled into 0.15.0 at recommendation of @bradleyrichter 

cc: @alexwykoff

**Comment by bsclifton** [comment]
Moving back to 0.15.1. There are some other changes which this relies on. Unfortunately, pulling in is not as easy as I thought ☹️

**Comment by luixxiul** [comment]
@bsclifton Was this merged to the latest build? if so would you please re-add `release-notes/include` for me? thanks!

**Comment by bsclifton** [comment]
@luixxiul it sure was- will be sure to add that label 😄
